2022_10_18

Kukuru-Dza opened the pull request:
binary search #1

Kukuru-Dza committed:
Update first_lessons_dushanbe/array_sorting_changing_values.py

Co-authored-by: shuvava <shuvava@users.noreply.github.com>

Commit:
Update first_lessons_dushanbe/array_sorting_changing_values.py

Co-authored-by: shuvava <shuvava@users.noreply.github.com>

Commit:
Merge remote-tracking branch 'origin/2022_10_13' into 2022_10_13

Commit:
Update first_lessons_dushanbe/array_sorting_changing_values.py

Co-authored-by: shuvava <shuvava@users.noreply.github.com>

Commit:
found mistake in array_sort_bubbles
tryng to implement binary search

Commit:
trying to implement binary search

Commit:
trying to implement binary search

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+        return -1
+    elif len(s_a) == 1:
+        return 0
+    elif i not in s_a:
```
you doing search two times. Operation `i not in s_a` is actually takes each element of `s_a` and compare with `i` which is O(N) complexity

Kukuru-Dza replied:
поменял

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+def bin_index(s_a, i):
+    if i is None or s_a is None:
+        return -1
+    elif len(s_a) == 1:
```
Could you please explain why you added this check?

Kukuru-Dza replied:
поменял

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+
+    first = 0
+    last = len(s_a) - 1
+    mid = (first + last) // 2
```
Why do you define `mid` twice here and on line 39? Could you simplify your code and define it once?

Kukuru-Dza replied:
поменял

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+        ln = 0
+        for ln in range(count):
+            if a[ln] > a[ln+1]:
+                (a[ln], a[ln + 1]) = (a[ln + 1], a[ln])
```
Could you please remove unnecessary `()`?

Kukuru-Dza replied:
done

shuvava commented on first_lessons_dushanbe/array_sorting_bubbles.py:
```diff
@@ -6,14 +6,14 @@ def list_sort(a):
         for ln in range(count):
             if a[ln] > a[ln+1]:
-                (a[ln+1], a[ln]) = (a[ln], a[ln+1])
+                (a[ln], a[ln + 1]) = (a[ln + 1], a[ln])
```
Could you please remove unnecessary `()?

Kukuru-Dza replied:
done

shuvava commented on first_lessons_dushanbe/array_sorting_bubbles.py:
```diff
@@ -6,14 +6,14 @@ def list_sort(a):
         return a
     count = len(a)-1
-    while count > 1:
+    while count >= 1:
```
you change condition, it means you did find a bug in your code. Could you please add test case for this bug?

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+    return a
+
+
+def bin_index(s_a, i):
```
Could you please add tests for this func?

Kukuru-Dza replied:
пытаюсь но у меня пока не работает корректно сама функция. она возвращает непонятно что при наличии повторяющихся элементов массива. может быть я вообще сделал этот поиск неправильно?

shuvava commented on first_lessons_dushanbe/array_sorting_binary_search.py:
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+
+
+def list_sort(a):
```
Why you add this method instead of using implementation from other file by referencing with `from file import func_name` ?

Kukuru-Dza replied:
done

Commit:
trying to implement binary search

Commit:
trying to implement binary search

Commit:
BINARY SEARCH WORKS FINE WITH ALL DIFFERENT NUMBERS IN ARRAY, BUT DOESN'T WORK CORRECTLY WITH REPEATED NUMBER

TEST NOT READY.

Commit:
BUT

All the versions from the wiki or geekbench doesnt work correctly with arrays like [1, 2, 3, 3, 3, 4, 5}

TEST NOT READY.

Commit:
BUT

All the versions from the wiki or geekbench doesnt work correctly with arrays like [1, 2, 3, 3, 3, 4, 5}

TEST NOT READY.

Commit:
BUT

All the versions from the wiki or geekbench doesnt work correctly with arrays like [1, 2, 3, 3, 3, 4, 5}

TEST NOT READY.
